dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -667,24 +667,25 @@ HRESULT SHELL32_GetFSItemAttributes(IShellFolder * psf, LPCITEMIDLIST pidl, LPDW
 
 HRESULT CFSFolder::_ParseSimple(
     _In_ LPOLESTR lpszDisplayName,
-    _Out_ WIN32_FIND_DATAW *pFind,
+    _Inout_ WIN32_FIND_DATAW *pFind,
     _Out_ LPITEMIDLIST *ppidl)
 {
     HRESULT hr;
     LPWSTR pchNext = lpszDisplayName;
 
     *ppidl = NULL;
 
+    const DWORD finalattr = pFind->dwFileAttributes;
+    const DWORD finalsizelo = pFind->nFileSizeLow;
     LPITEMIDLIST pidl;
     for (hr = S_OK; SUCCEEDED(hr); hr = SHILAppend(pidl, ppidl))
     {
         hr = Shell_NextElement(&pchNext, pFind->cFileName, _countof(pFind->cFileName), FALSE);
         if (hr != S_OK)
             break;
 
-        if (pchNext)
-            pFind->dwFileAttributes = FILE_ATTRIBUTE_DIRECTORY;
-
+        pFind->dwFileAttributes = pchNext ? FILE_ATTRIBUTE_DIRECTORY : finalattr;
+        pFind->nFileSizeLow = pchNext ? 0 : finalsizelo;
```

Comment:
Shouldn't you also set the `nFileSizeHigh` to a consistent value too?

Reply:
Technically, yes but.

- We don't actually store the upper 32-bits inside the pidl, only NT6 does that.
- CFSFolder needs its entire pidl handling to be rewritten and not rely on wine\pidl.c

Reply:
thanks, i'll approve the PR then.

Reply:
I don't see the need for a comment. Wine\pidl.h defines two structs used by the filesystem pidls, neither of them have a member for the high size32.